sdk-cpp/tests/dosdkcpp_testrunner.cpp

```diff
@@ -11,15 +16,26 @@ int main(int argc, char** argv)
     po::options_description desc("Options");
     desc.add_options()
         ("help,h", "Show help messages")
-        ("mcc-host,m", po::value<std::string>(), "MCC hostname to use (optional override)");
+        ("mcc-host,m", po::value<std::string>(), "MCC hostname to use (optional override)")
+        ("manual-start,d", po::bool_switch()->default_value(false), "Wait for command line input to start tests (useful for debugging)");
     po::variables_map vm;
     po::store(po::parse_command_line(argc, argv, desc), vm);
+
     auto it = vm.find("mcc-host");
     if (it != vm.end())
     {
         g_mccHostName = it->second.as<std::string>();
         std::cout << "Got overriden MCC host: " << g_mccHostName << '\n';
     }
 
+    auto manualStart = vm["manual-start"].as<bool>();
+    if (manualStart)
+    {
+        do
+        {
+            std::cout << '\n' << "Press a key to continue...";
```

Comment:
You could start the test under the debugger instead of this. Just FYI.

Reply:
I normally start the exe and attach to process - is there a way to launch the exe and immediately attach?

---
In reply to: [612797218](https://github.com/microsoft/do-client/pull/24#discussion_r612797218) [](ancestors = 612797218)

Reply:
Linux, use gdb ./deliveryoptimization_sdk_tests and include the --args option if you are passing args to the debuggee.
Windows, you can launch an executable from within windbg. Should be possbile with VS as well but it's been a while since I used it.

---
In reply to: [612800145](https://github.com/microsoft/do-client/pull/24#discussion_r612800145) [](ancestors = 612800145,612797218)

Reply:
Gotcha, Yeah I usually use the VS debugger - I'll leave as is